pkg/store/prometheus.go

```diff
@@ -220,6 +225,55 @@ func (p *PrometheusStore) Series(r *storepb.SeriesRequest, s storepb.Store_Serie
 	return p.handleStreamedPrometheusResponse(s, httpResp, queryPrometheusSpan, extLset)
 }
 
+func (p *PrometheusStore) queryPrometheus(s storepb.Store_SeriesServer, r *storepb.SeriesRequest) error {
+	ctx := context.Background()
+	queryOpts := promclient.QueryOptions{
+		Deduplicate:             false,
+		PartialResponseStrategy: storepb.PartialResponseStrategy_WARN,
+		Method:                  "GET",
+	}
+
+	matrix, _, err := p.client.QueryRange(ctx, p.base, r.ToPromQL(), r.MinTime, r.MaxTime, r.QueryHints.Step/1000, queryOpts)
```

Comment:
Is it always range query?

Reply:
I think it needs to be because of lookback, but I can double check.

Reply:
Looks like we can always use a query range, we just need to use a positive step for instant queries.